spring-boot-project/spring-boot/src/main/java/org/springframework/boot/ssl/jks/JksSslStoreBundle.java

```diff
@@ -109,14 +111,29 @@ private void loadHardwareKeyStore(KeyStore store, String location, char[] passwo
 	private void loadKeyStore(KeyStore store, String location, char[] password) {
 		Assert.state(StringUtils.hasText(location), () -> "Location must not be empty or null");
 		try {
-			URL url = ResourceUtils.getURL(location);
-			try (InputStream stream = url.openStream()) {
-				store.load(stream, password);
+			if (isBase64(location.getBytes())) {
```

Comment:
Maybe instead of doing the decode twice, where you test first if you can decode it, you could do the else part in the catch block?